tests/test_dcrAssembleCoadd.py

```diff
@@ -0,0 +1,295 @@
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+
+from collections import namedtuple
+from astropy import units as u
+import numpy as np
+import scipy.ndimage.interpolation
+from scipy.ndimage.morphology import binary_dilation as dilate
+import unittest
+
+from lsst.afw.coord import Coord, IcrsCoord, Observatory, Weather
+import lsst.afw.geom as afwGeom
+from lsst.afw.geom import Angle, makeCdMatrix, makeSkyWcs
+import lsst.afw.image as afwImage
+import lsst.utils.tests
+from lsst.pipe.tasks.dcrAssembleCoadd import DcrAssembleCoaddTask, DcrAssembleCoaddConfig
+
+
+class DcrAssembleCoaddTestTask(lsst.utils.tests.TestCase):
+    """! A test case for the DCR-aware image coaddition algorithm.
+    """
+
+    def setUp(self):
+        self.config = DcrAssembleCoaddConfig()
+        self.config.dcrNSubbands = 3
+        self.config.lambdaEff = 478.
+        self.config.filterWidth = 147.
+        self.config.filterName = 'g'
+        badMaskPlanes = self.config.badMaskPlanes[:]
+        badMaskPlanes.append("CLIPPED")
+        size = 40
+        psfSize = 2
+        nSrc = 5
+        seed = 5
+        self.randGen = np.random
+        self.randGen.seed(seed)
+        noiseLevel = 5
+        sourceSigma = 20.
+        fluxRange = 2.
+        edgeDist = self.config.bufferSize
+        xLoc = self.randGen.random(nSrc)*(size - 2*edgeDist) + edgeDist
+        yLoc = self.randGen.random(nSrc)*(size - 2*edgeDist) + edgeDist
+        self.dcrModels = []
+        self.bbox = afwGeom.Box2I(afwGeom.Point2I(0, 0), afwGeom.Extent2I(size, size))
+        self.pixelScale = 0.2*afwGeom.arcseconds
+
+        imageSum = np.zeros((size, size))
+        for subfilter in range(self.config.dcrNSubbands):
+            flux = (self.randGen.random(nSrc)*(fluxRange - 1.) + 1.)*sourceSigma*noiseLevel
+            model = afwImage.MaskedImageF(self.bbox)
+            image = model.getImage().getArray()
+            image += self.randGen.random((size, size))*noiseLevel
+            for x, y, f in zip(xLoc, yLoc, flux):
+                xVals = np.exp(-np.power(np.arange(size) - x, 2.)/(2*np.power(psfSize, 2.)))
+                yVals = np.exp(-np.power(np.arange(size) - y, 2.)/(2*np.power(psfSize, 2.)))
+                image += f*np.outer(yVals, xVals)
```

Comment:
We have multiple codes for creating PSF images in the stack. Can you use one of them instead of writing something new?

Reply:
Could you please point me to code you would recommend I use in the stack?

Reply:
You prolly want [this](https://github.com/lsst/meas_algorithms/blob/master/python/lsst/meas/algorithms/testUtils.py#L32), but there are [more complicated versions](https://github.com/lsst/pipe_tasks/blob/master/python/lsst/pipe/tasks/mocks/mockCoadd.py#L94) also.